### ISSUE TITLE
Is this an error in the prompt optimization colab?

### ISSUE THREAD

**Issue by nsadeh** [zou-group/textgrad#135]
The colab for prompt optimization has the following function:

```
def run_validation_revert(system_prompt: tg.Variable, results, model, eval_fn, val_set):
    val_performance = np.mean(eval_dataset(val_set, eval_fn, model))
    previous_performance = np.mean(results["validation_acc"][-1])
    print("val_performance: ", val_performance)
    print("previous_performance: ", previous_performance)
    previous_prompt = results["prompt"][-1]
    
    if val_performance < previous_performance:
        print(f"rejected prompt: {system_prompt.value}")
        system_prompt.set_value(previous_prompt)
        val_performance = previous_performance

    results["validation_acc"].append(val_performance)
 ```
 
 We measure the performance in terms of loss, where lower is better, so shouldn't the inequality sign flip here?
 
 My apologies if this is silly, I am new to ML.

**Comment by vinid** [comment]
@mertyg can correct me if I am wrong, but this should be correct:

if the new prompt (t+1) does not improve performance we are going to discard this and try optimization again using the most recent prompt (t).

What's happening behind the scenes is that after he `optimizer.step()` instruction we update the system prompt. So what we do in the above function is to revert it in case the update was not good enough.

**Comment by nsadeh** [comment]
OK, but isn't val_performance = performance of new system prompt and previous_performance = performance of previous system prompt? If so, then `val_performance < previous_performance` is equivalent to saying "if the loss of the new system prompt is less than the loss of the previous system prompt." Doesn't that mean it's better, and so should be kept, not rejected?

I ran a version of this adapted to my prompts and when the inequality was unchanged, the loss stayed the same, and when I reversed the inequality the loss went down.

**Comment by vinid** [comment]
That would be right if we were evaluating on a loss, but in this case we are looking at accuracy.

So lower accuracy in validation means worse prompt and thus rejection.

**Comment by nsadeh** [comment]
Oh it looks like I mixed the 1 and 0 up on the prompt (https://github.com/zou-group/textgrad/blob/204266399fb298ac69d4fd5ce6b94c51acd95046/textgrad/tasks/__init__.py#L58). I thought it would be 0 for accurate and 1 for inaccurate to preserve the same directionality as loss.

In my adapted code it is that way, so that's why I was seeing the wrong results with the inequality. Thanks for bearing with me and clearing that up.
